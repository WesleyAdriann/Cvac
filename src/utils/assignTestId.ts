import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
