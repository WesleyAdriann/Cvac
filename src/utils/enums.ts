export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
