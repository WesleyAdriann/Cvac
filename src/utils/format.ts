export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
