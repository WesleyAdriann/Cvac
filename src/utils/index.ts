export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
