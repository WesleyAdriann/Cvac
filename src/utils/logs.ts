export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
