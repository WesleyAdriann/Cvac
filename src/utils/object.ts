export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
