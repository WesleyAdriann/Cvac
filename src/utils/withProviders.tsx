import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
