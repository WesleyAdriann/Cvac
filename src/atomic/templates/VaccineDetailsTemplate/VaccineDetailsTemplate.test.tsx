import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '../../../utils'

import { VaccineDetailsTemplate, IVaccineDetailsTemplate } from './index'

describe('Templates/VaccineDetailsTemplate', () => {
  const props: IVaccineDetailsTemplate = {
    testID: 'VaccineDetailsTemplate'
  }

  const elContainer = mockTestID('SafeAreaView', props.testID!).testID

  const renderComponent = () => render(<VaccineDetailsTemplate {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
