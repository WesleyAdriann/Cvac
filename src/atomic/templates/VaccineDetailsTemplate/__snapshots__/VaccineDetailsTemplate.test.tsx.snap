// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/VaccineDetailsTemplate should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={false}
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    testID="SafeAreaView_VaccineDetailsTemplate"
  >
    <View
      accessible={false}
      flex={1}
      paddingStyle={16}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
      testID="View_Flex"
    >
      <View
        accessible={false}
        flex={0.5}
        justifyContent="center"
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.5,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
        testID="View_Flex"
      >
        <Text
          accessible={false}
          align="center"
          size={30}
          style={
            Array [
              Object {
                "color": "#424242",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#424242",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginBottom": 0,
                  "textAlign": "center",
                  "textDecorationColor": "black",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                },
              ],
            ]
          }
          testID="Text_Text"
        >
          Febre Amarela
        </Text>
      </View>
      <RCTScrollView
        accessible={false}
        contentContainerStyle={
          Object {
            "flexGrow": 1,
            "padding": undefined,
          }
        }
        flex={2}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 2,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
        testID="ScrollView_Flex"
      >
        <View>
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
            testID="View_Flex"
          >
            <Text
              accessible={false}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 0,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
              weight="bold"
            >
              Doses
            </Text>
            <Text
              accessible={false}
              marginBottom={8}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "marginBottom": 8,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
            >
              3
            </Text>
            <Text
              accessible={false}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 0,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
              weight="bold"
            >
              Quando
            </Text>
            <Text
              accessible={false}
              marginBottom={8}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "marginBottom": 8,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
            >
              1 meses
2 meses
3 meses
              
            </Text>
            <Text
              accessible={false}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 0,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
              weight="bold"
            >
              Prevenção
            </Text>
            <Text
              accessible={false}
              marginBottom={8}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "marginBottom": 8,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
            >
              Febre Amarela
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
